crates/examples/push-cdn/marshal.rs

```diff
@@ -0,0 +1,53 @@
+//! The following is the main `Marshal` binary, which just instantiates and runs
+//! a `Marshal` object with the `HotShot` types.
+//!
+use anyhow::{Context, Result};
+use cdn_broker::reexports::connection::protocols::Quic;
+use cdn_marshal::{ConfigBuilder, Marshal};
+use clap::Parser;
+use hotshot::traits::implementations::WrappedSignatureKey;
+use hotshot_example_types::node_types::TestTypes;
+use hotshot_types::traits::node_implementation::NodeType;
+
+//TODO: for both client and marshal, clean up and comment `main.rs`
+// TODO: forall, add logging where we need it
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+/// The main component of the push CDN.
+struct Args {
+    /// The discovery client endpoint (including scheme) to connect to
+    #[arg(short, long)]
+    discovery_endpoint: String,
```

Comment:
Is this intended to be something like an `http://` type value? The comment seems to hint at that, but perhaps an example in the doc comment or help text could clarify.

Reply:
It is an `IP:port` pairing, but this makes a lot of sense. I will provide an example in the comment